specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/preview/2022-09-01-preview/vaults.json

```diff
@@ -0,0 +1,2128 @@
+      "post": {
+        "tags": [
+          "RecoveryServices"
+        ],
+        "summary": "API to check for resource name availability.\r\nA name is available if no other resource exists that has the same SubscriptionId, Resource Name and Type\r\nor if one or more such resources exist, each of these must be GC'd and their time of deletion be more than 24 Hours Ago",
+        "operationId": "RecoveryServices_CheckNameAvailability",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionId"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupName"
+          },
+          {
+            "$ref": "#/parameters/ApiVersion"
+          },
+          {
+            "name": "location",
+            "in": "path",
+            "description": "Location of the resource",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "input",
+            "in": "body",
+            "description": "Contains information about Resource type and Resource name",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/CheckNameAvailabilityParameters"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/CheckNameAvailabilityResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Availability status of Resource Name when no resource with same name, type and subscription exists, nor has been deleted within last 24 hours": {
+            "$ref": "./examples/CheckNameAvailability_Available.json"
+          },
+          "Availability status of Resource Name when resource with same name, type and subscription exists": {
+            "$ref": "./examples/CheckNameAvailability_NotAvailable.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.RecoveryServices/locations/{location}/capabilities": {
+      "post": {
+        "tags": [
+          "RecoveryServices"
+        ],
+        "summary": "API to get details about capabilities provided by Microsoft.RecoveryServices RP",
+        "operationId": "RecoveryServices_Capabilities",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionId"
```

Comment:
>SubscriptionId [](http://example.com/codeflow?start=34&length=14)

Consider getting this and ApiVersion from v3 common types.

Reply:
Thanks. Now using SubscriptionId, ResourceGroupName and ApiVersion from v3 common types. However, this resulted in having to rename 'Resource' and 'TrackedResource' class names in definitions to 'RecoveryServicesResource' and 'RecoveryServicesTrackedResource' respectively as they were not getting recognized without the renaming

Reply:
Awesome, thank-you: this is a good change.